### PR TITLE
chore: bump blocksuite

### PR DESCRIPTION
**Pull request by fundon**
### Patch Changes

- [3c61be5](https://github.com/toeverything/blocksuite/commit/3c61be5dedcc1fc37d6f09ed2541b391584c0ded): - Refactor drag handle widget
  - Split embed blocks to `@blocksuite/affine-block-embed`
  - Fix latex selected state in edgeless mode
  - Fix unclear naming
  - Fix prototype pollution
  - Fix portal interaction in affine modal
  - Fix paste linked block on edgeless
  - Add scroll anchoring widget
  - Add highlight selection

### CODE REVIEW

**Comment by fundon** [comment]
* **#8230** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8230?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2  dependent PRs ([#8253](https://github.com/toeverything/AFFiNE/pull/8253) <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8253?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#8269](https://github.com/toeverything/AFFiNE/pull/8269) <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8269?utm_source=stack-comment-icon" target="_blank">Graphite</a>) 👈
* **#8227** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8227?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @fundon and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by pengx17 — APPROVED** [review]
can you please also add a new e2e test?

**Comment by fundon** [comment]
> can you please also add a new e2e test?

👌

File: packages/frontend/core/src/modules/quicksearch/impls/links.ts [hunk 1]
```diff
@@ -31,11 +31,29 @@ export class LinksQuickSearchSession
...
-    if (!isLink) return [];
+    let isLink = false;
+
+    try {
+      let url = query;
+      const hasScheme = ['http://', 'https://'].some(s => url.startsWith(s));
+
+      if (!hasScheme) {
+        const dotIdx = url.indexOf('.');
+        if (dotIdx > 0 && dotIdx < url.length - 1) {
```
Comment (EYHN) [inline packages/frontend/core/src/modules/quicksearch/impls/links.ts]:
```suggestion
        if (dotIdx !== -1) {
```

File: packages/frontend/core/src/modules/quicksearch/impls/links.ts [hunk 2]
```diff
@@ -31,11 +31,29 @@ export class LinksQuickSearchSession
...
+    const result: QuickSearchItem<'link', LinkPayload>[] = [];
+    if (!query) return result;
 
-    const isLink = query.startsWith('http://') || query.startsWith('https://');
-    if (!isLink) return [];
+    let isLink = false;
+
+    try {
+      let url = query;
+      const hasScheme = ['http://', 'https://'].some(s => url.startsWith(s));
```
Comment (EYHN) [inline packages/frontend/core/src/modules/quicksearch/impls/links.ts]:
```suggestion
      const hasScheme = url.match(/^https?:\/\//)
```

File: packages/frontend/core/src/modules/quicksearch/impls/links.ts [hunk 3]
```diff
@@ -31,11 +31,29 @@ export class LinksQuickSearchSession
...
   items$ = LiveData.computed(get => {
-    const query = get(this.query$);
-    if (!query) return [];
+    const query = get(this.query$).trim();
+    const result: QuickSearchItem<'link', LinkPayload>[] = [];
+    if (!query) return result;
 
-    const isLink = query.startsWith('http://') || query.startsWith('https://');
-    if (!isLink) return [];
+    let isLink = false;
```
Comment (EYHN) [inline packages/frontend/core/src/modules/quicksearch/impls/links.ts]: turn this to a util function

File: packages/frontend/core/src/modules/quicksearch/impls/links.ts (hunk 1 above)
Comment (fundon) [reply packages/frontend/core/src/modules/quicksearch/impls/links.ts]: When dot is at the end, it is not recognized as domian

File: packages/frontend/core/src/modules/quicksearch/impls/links.ts (hunk 3 above)
Comment (EYHN) [reply packages/frontend/core/src/modules/quicksearch/impls/links.ts:38]: we need utils `isLink` `isInternalLink` `isExternalLink`

File: packages/frontend/core/src/modules/navigation/utils.ts [hunk 4]
```diff
@@ -61,6 +63,23 @@ export const resolveRouteLinkMeta = (href: string) => {
   }
 };
 
+export const isExternalLink = (href: string) => {
```
Comment (EYHN) [inline packages/frontend/core/src/modules/navigation/utils.ts]: This feels too lenient; any string containing a `.` will pass this test.

Comment (EYHN) [inline packages/frontend/core/src/modules/navigation/utils.ts]: i think rename this to `isLink`

**Review by EYHN — APPROVED** [review]

**Pull request merged**